### PR TITLE
ZONE module

### PR DESCRIPTION
**Pull request by bvandersloot**
Performs ALOOKUP on all domains in a root zone file. This attempts to skip the first hop where possible, not hitting local or root DNS servers. It instead uses glue records and NS information in the root zone file to connect directly to the nameserver.

This is not general, and currently only works with ALOOKUP.

### CODE REVIEW

File: lookup.go [hunk 1]
```diff
@@ -73,17 +73,35 @@ func doLookup(g *GlobalLookupFactory, gc *GlobalConf, input <-chan string, outpu
...
+			prefixedName, changed := makeName(rawName, gc.NamePrefix)
+			if changed {
+				res.AlteredName = prefixedName
+				targeted.Domain = prefixedName
+			}
+			query, err := json.Marshal(targeted)
+			lookupName = string(query)
+			if err != nil {
+				log.Fatal("Internal Error: %s", err)
+			}
```
Comment (zakird) [inline lookup.go:90]: Why is this in the framework and not in the module itself?

File: conf.go [hunk 2]
```diff
@@ -51,6 +53,11 @@ type Result struct {
 	Data        interface{} `json:"data,omitempty"`
 }
 
+type TargetedDomain struct {
+	Domain      string   `json:"domain"`
+	Nameservers []string `json:"nameservers"`
+}
+
```
Comment (zakird) [inline conf.go:60]: Does this need to be defined globally?

File: lookup.go [hunk 3]
```diff
@@ -140,12 +158,56 @@ func doInput(in chan<- string, path string, wg *sync.WaitGroup) error {
...
+			in <- string(query)
+		}
+	} else {
+		s := bufio.NewScanner(f)
+		for s.Scan() {
+			in <- s.Text()
+		}
+		if err := s.Err(); err != nil {
+			log.Fatal("input unable to read file", err)
+		}
```
Comment (zakird) [inline lookup.go:210]: Same question as earlier---why is this in the framework and not in the zone module itself?

**Comment by bvandersloot** [comment]
Looking to make changes to make the zonefile input option more generic.

**Comment by zakird** [comment]
@bvandersloot I'm going to close this given our conversation since I don't want someone to spend time to review given the rearrange. Let's just open a new PR when you're ready?

**Pull request closed without merging**